tests/main/mount-ns/task.yaml

```diff
@@ -0,0 +1,112 @@
+summary: The shape of the mount namespace on classic systems for non-classic snaps
+systems: [ubuntu-16.04-64, ubuntu-18.04-64, ubuntu-core-16-64, ubuntu-core-18-64]
+# The test is sensitive to backend type, which designates the used image.
+# Backends are enabled one-by-one along with the matching data set.
+backends: [google]
+details: |
+    This test measures the mount table of the host, of the mount namespace for
+    a simple core16-based snap for the per-user mount namespace of a simple
+    core16-base snap as well as for a simple core18 based snap and finally of a
+    simple snap using classic confinement.
+
+    The mount tables are obtained from /proc/pid/mountinfo interface. They are
+    then fed to mountinfo-tool with various determinism adjustment options and
+    compared to stock tables created when the test was first made.
+
+    Naturally occurring mount tables will differ from invocation to invocation
+    for several reasons. Snap revisions seen in various paths will drift over
+    time. The order of mount operations that can be done in parallel will
+    differ from one boot to another. Good examples of that include
+    /snap/name/revision, which can be all started in parallel. Another one is
+    the set of control groups mounted in sysfs. Some mount options may be
+    created based on the amount of available memory. Some mount entries may
+    contain numeric IDs that are allocated and are hard to predict. Block
+    devices backing actual file systems may be on various disks, for instance
+    on /dev/vda or /dev/sdb. All of those are handled by mountinfo-tool
+    --rename and --renumber, along with sorting options that combat
+    non-deterministic mount order.
+
+    Unfortunately this test need separate data sets for google compute engine
```

Comment:
The initial mount namespace differs between test backends. This is captured by grouping the test data by backend type.